pkg/controller/reconciler/recipe_webhook_test.go

```diff
@@ -0,0 +1,248 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconciler
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+	"strings"
+	"testing"
+	"time"
+
+	radappiov1alpha3 "github.com/radius-project/radius/pkg/controller/api/radapp.io/v1alpha3"
+	portableresources "github.com/radius-project/radius/pkg/rp/portableresources"
+	"github.com/radius-project/radius/test/testcontext"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+const (
+	defaultNamespace    = "default"
+	validResourceType   = "Applications.Core/extenders"
+	invalidResourceType = "Applications.Core/invalidType"
+)
+
+// Test_ValidateRecipe_Type tests a recipe with valid and invalid types.
+func Test_ValidateRecipe_Type(t *testing.T) {
+	ctx := testcontext.New(t)
+	radius, client := setupWebhookTest(t)
+
+	// Environment is created.
+	createEnvironment(radius, "default")
+
+	t.Run("test recipe for invalid type", func(t *testing.T) {
+		recipeName := "test-recipe-invalidtype"
+		namespace := types.NamespacedName{Namespace: defaultNamespace, Name: recipeName}
+		recipe := makeRecipe(namespace, invalidResourceType)
+
+		err := client.Create(ctx, &corev1.Namespace{ObjectMeta: ctrl.ObjectMeta{Name: namespace.Name}})
+		require.NoError(t, err)
+
+		// We do not check for error here since error may be captured after this initial call is completed.
+		_ = client.Create(ctx, recipe)
```

Comment:
Can you explain this? If the type is invalid, I'd expect this to return an error. Isn't this the thing we're trying to check?

Reply:
Thank you! Updated code to catch the error. 
Also pls review new function ln 309: [updateWebhookFailurePolicy()](https://github.com/radius-project/radius/blob/8519fd58a053fb3970287efd01abcfc5b07f63d5/pkg/controller/reconciler/recipe_webhook_test.go#L302) . I'm using it to set webhook FailurePolicy to Fail when running Webhook tests and then setting it back to Ignore after tests are run, so it does not interfere with other reconciler tests.